adagrams/game.py

```diff
@@ -1,11 +1,170 @@
+temp_list = []
+
 def draw_letters():
-    pass
 
-def uses_available_letters(word, letter_bank):
-    pass
+    # Creates a comprehensive list which includes the total quanity of each available letter from the LETTER_POOL dictionary as individual elements.  
+    for letter, quantity in LETTER_POOL.items():
+        temp_list.append([letter] * quantity)
+ 
+    for sublist in temp_list:
+        for letter in sublist:
+            LETTER_POOL_LIST.append(letter)
+
+    # Chooses 10 random letters as hand, excluding the ones that are already chosen.
+    #hand = random.sample(LETTER_POOL_LIST, 10)
+
+    hand = []
+    while len(hand) < 10:
+        random_letter = random.choice(LETTER_POOL_LIST)
+        random_letter_index = LETTER_POOL_LIST.index(random_letter)
+        LETTER_POOL_LIST.pop(random_letter_index)
+        hand.append(random_letter)
+
+    return hand
+
+draw_letters()
+     
+def uses_available_letters(word, letter_bank):       
+    word_verification = []
+    hand_letter_count = {}
+    word_letter_count = {}
+    count = 0
+    capitalized_word = word.upper()
+
+    for letter in letter_bank:
+        if letter not in hand_letter_count:
+            count = 1
+            hand_letter_count[letter] = count
+        else: 
+            count += 1
+            hand_letter_count[letter] = count
+      
+    for letter in capitalized_word:
+        letter_quantity = capitalized_word.count(letter)
+        word_letter_count[letter] = letter_quantity
+
+        if letter in letter_bank and word_letter_count[letter] <= hand_letter_count[letter]:
+            word_verification.append("True")
+        else:
+            word_verification.append("False")
+      
+    if "False" in word_verification:
+        return False
+    else:
+        return True
+      
 
 def score_word(word):
-    pass
+    word_score = {}
+
+    if word == "":
+       word_score["empty"] = 0
+
+    for letter in word.upper():
+        if letter in SCORE_CHART:
+            if letter not in word_score:
+                word_score[letter] = SCORE_CHART[letter]
+            else:
+                word_score[letter] += SCORE_CHART[letter]
+
+    if 7 <= len(word) <= 10:
+        word_score["bonus"] = 8
+      
+    return sum(word_score.values())
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    score_dict = {}
+    score_list = []
+    matching_scores_dict = {}
+    highest_score = 0
+
+    for word in word_list:
+        score = score_word(word)
+        score_dict[word] = score
+
+    for word, score in score_dict.items():
+        score_list.append(score)
+        if score > highest_score:
+            highest_score = score
+            winning_word = word
+    
+    matching_scores = [k for k,v in Counter(score_list).items() if v>1]
+
+    for word, score in score_dict.items():
+        if score in matching_scores:
+            matching_scores_dict[word] = score
+
+    for word, score in matching_scores_dict.items():
```

Comment:
Instead of iterating through `matching_scores.dict.items()`, what if we iterate through `score_dict` instead? `score_dict` is a dictionary that already keeps each word and its score. We already have a value for `highest_score` and `winning_word`, too!

If we refactor this loop so it iterates through `score_dict`, we can also delete the code around matching scores. If we make these changes, then the tests still pass!